restapi.go

```diff
@@ -233,26 +235,24 @@ func (self *RestAPI) pushNotification(reqId string, kv map[string]string, perdp
 		case "service":
 			// three keys need to be ignored
 		case "badge":
-			if v != "" {
-				var e error
-				_, e = strconv.Atoi(v)
-				if e == nil {
-					notif.Data["badge"] = v
-				} else {
-					notif.Data["badge"] = "0"
-				}
+			if _, err := strconv.Atoi(v); err == nil {
+				notif.Data["badge"] = v
+			} else {
+				notif.Data["badge"] = "0"
 			}
 		default:
 			notif.Data[k] = v
 		}
 	}
 
+	logger.Debugf("Parameters accepted: %+v", notif)
+
 	if notif.IsEmpty() {
-		logger.Errorf("RequestId=%v From=%v Service=%v EmptyNotification", reqId, remoteAddr, service)
+		logger.Errorf("RequestId=%v From=%v Service=%v NrSubscribers=%v Subscribers=\"%+v\".", reqId, remoteAddr, service, len(subs), subs, err)
```

Comment:
I can't argue with that logic. It makes sense to be explicit.

Tina
On Jun 9, 2015 10:58 AM, "Misha Nasledov" <notifications@github.com> wrote:

> In restapi.go
> <https://github.com/uniqush/uniqush-push/pull/66#discussion_r32043226>:
>
> >  	if notif.IsEmpty() {
> > -		logger.Errorf("RequestId=%v From=%v Service=%v EmptyNotification", reqId, remoteAddr, service)
> > +		logger.Errorf("RequestId=%v From=%v Service=%v NrSubscribers=%v Subscribers=\"%+v\".", reqId, remoteAddr, service, len(subs), subs, err)
>
> I like how it's explicit about what the problem is versus the fact it's
> just an error
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/uniqush/uniqush-push/pull/66/files#r32043226>.
>